test/sslapitest.c

```diff
@@ -3288,6 +3288,111 @@ static int test_ciphersuite_change(void)
+        TLS1_3_RFC_AES_128_CCM_8_SHA256":"TLS1_3_RFC_AES_128_CCM_SHA256};
+    const char *t13_cipher = NULL;
+    const char *t12_cipher = NULL;
+    int set_at_ctx = 0;
+    int set_at_ssl = 0;
+    int i;
+
+    switch (idx) {
+        case 0:
+            set_at_ctx = 1;
+            break;
+        case 1:
+            set_at_ssl = 1;
+            break;
+        case 2:
+            set_at_ctx = 1;
+            t12_cipher = TLS1_TXT_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256;
+            break;
+        case 3:
+            set_at_ssl = 1;
+            t12_cipher = TLS1_TXT_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256;
+            break;
+    }
+
+    for (i = 0; i < (int)(sizeof(t13_ciphers)/sizeof(t13_ciphers[0])); i++) {
+        t13_cipher = t13_ciphers[i];
+        if (!TEST_true(create_ssl_ctx_pair(TLS_server_method(), TLS_client_method(),
+                                           TLS1_VERSION, TLS_MAX_VERSION,
+                                           &sctx, &cctx, cert, privkey)))
+            goto end;
+
+        if (set_at_ctx) {
+            if (t13_cipher && (!TEST_true(SSL_CTX_set_ciphersuites(sctx, t13_cipher))))
+                goto end;
+            if (t13_cipher && (!TEST_true(SSL_CTX_set_ciphersuites(cctx, t13_cipher))))
+                goto end;
+            if (t12_cipher && (!TEST_true(SSL_CTX_set_cipher_list(sctx, t12_cipher))))
+                goto end;
+            if (t12_cipher && (!TEST_true(SSL_CTX_set_cipher_list(cctx, t12_cipher))))
+                goto end;
+        }
+
+        if (!TEST_true(create_ssl_objects(sctx, cctx, &serverssl, &clientssl,
+                                                 NULL, NULL)))
+            goto end;
+
+        if (set_at_ssl) {
+            if (t13_cipher && (!TEST_true(SSL_set_ciphersuites(serverssl, t13_cipher))))
+                goto end;
+            if (t13_cipher && (!TEST_true(SSL_set_ciphersuites(clientssl, t13_cipher))))
+                goto end;
+            if (t12_cipher && (!TEST_true(SSL_set_cipher_list(serverssl, t12_cipher))))
+                goto end;
+            if (t12_cipher && (!TEST_true(SSL_set_cipher_list(clientssl, t12_cipher))))
+                goto end;
+        }
+
+        if (!TEST_true(create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE)))
+            goto end;
+
+        if (strcmp(SSL_CIPHER_get_name(SSL_get_current_cipher(serverssl)),
+                   SSL_CIPHER_get_name(SSL_get_current_cipher(clientssl))) != 0)
+            goto end;
+        if (strncmp(t13_cipher, SSL_CIPHER_get_name(SSL_get_current_cipher(serverssl)),
+                strlen(SSL_CIPHER_get_name(SSL_get_current_cipher(serverssl)))) != 0)
+            goto end;
+        if (strncmp(t13_cipher, SSL_CIPHER_get_name(SSL_get_current_cipher(clientssl)),
+                strlen(SSL_CIPHER_get_name(SSL_get_current_cipher(clientssl)))) != 0)
+            goto end;
+
+        SSL_free(serverssl);
```

Comment:
This isn't the usual error handling setup.

```
    ret = 1;
end:
    SSL_free(serverssl);
    SSL_free(clientssl);
    SSL_CTX_free(sctx);
    SSL_CTX_free(cctx);
    return ret;
```

With the addition of `int ret = 0;` at the start of the function.